addon/services/file-manager.js

```diff
@@ -136,30 +269,46 @@ export default Ember.Service.extend({
         return new Ember.RSVP.Promise((resolve, reject) => {
             Ember.$.ajax(url, {
                 method,
-                data,
+                data: options.data,
                 processData: false,
                 headers: {
                     Authorization: `Bearer ${accessToken}`
                 },
-                success: (data) => resolve(data),
-                error: (_, __, error) => reject(error)
+                error: (_, __, error) => reject(error),
+                success: (data) => resolve(data)
             });
         });
     },
 
-    _pushToStore(data, id, parentFolder) {
-        // Hack the Waterbutler entity to look like the file Ember expects
-        data.data.id = id;
-        data.data.type = 'file';
-        let attr = data.data.attributes;
-        attr.materializedPath = attr.materialized;
-        attr.dateModified = attr.modified;
-
-        let newFolder = this.get('store').push(data);
+    /**
+     * Force-reload a model from the API.
+     *
+     * @method _reloadModel
+     * @private
+     * @param {Object} model Any Ember model with a `reload` method.
+     * @return {Promise} Promise that resolves to the reloaded model or
+     * rejects with an error message.
+     */
+    _reloadModel(model) {
+        this.set('isReloading', true);
+        return model.reload().then((freshModel) => {
+            this.set('isReloading', false);
```

Comment:
I'll try to think about alternatives when I review this in depth later, but this is only one of many ways that shared state on a singleton could come back to bite us. If we can get away without the instance variable on the service, that would be strongly preferred. Just setting `isReloading` back doesn't change the fact that the variable exists.

For example: what happens if I make two different file requests in rapid succession? Would the second request pick up on the `isReloading` state belonging to the first request, and trigger a different behavior than intended? That would be very hard to debug.

The key is that state is being shared among unrelated requests, and not just across time for one particular use case.